src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -173,6 +173,29 @@ public KeyMaskingConfig getMaskConfigIfMatched(byte[] bytes, int keyOffset, int
         }
     }
 
+    /**
+     * An overload of {@code dev.blaauwendraad.masker.json.KeyMatcher#getMaskConfigIfMatched(byte[], int, int, java.util.Iterator)} that does only jsonpath lookup.
+     */
+    @CheckForNull
+    public KeyMaskingConfig getMaskConfigIfMatched(byte[] bytes, Iterator<SegmentReference> jsonPath) {
+        if (maskingConfig.isInMaskMode()) {
+            TrieNode node = searchForJsonPathKeyNode(bytes, jsonPath);
+            if (node != null && !node.negativeMatch) {
+                return node.keyMaskingConfig;
+            }
+            return null;
+        } else {
+            TrieNode node = searchForJsonPathKeyNode(bytes, jsonPath);
```

Comment:
Maybe add one line of doc

// in allow mode

Reply:
I will add more docs there.